wolfcrypt/src/port/atmel/atmel.c

```
@@ -0,0 +1,240 @@
+}
+
+
+/* Function to return slot number to avail list */
+void atmel_ecc_free(int slot)
+{
+    if (slot >= 0 && slot <= ATECC_MAX_SLOT) {
+        /* Mark slot of free */
+        mSlotList[slot] = ATECC_INVALID_SLOT;
+    }
+}
+
+
+/**
+ * \brief Give enc key to read pms.
+ */
+static int atmel_set_enc_key(uint8_t* enckey, int16_t keysize)
+{
+    if (enckey == NULL || keysize != ATECC_KEY_SIZE) {
+        return -1;
+    }
+
+    XMEMSET(enckey, 0xFF, keysize); // use default values
+
+    return SSL_SUCCESS;
+}
+
+
+/**
+ * \brief Write enc key before.
+ */
+static int atmel_init_enc_key(void)
+{
+	uint8_t ret = 0;
+	uint8_t read_key[ATECC_KEY_SIZE] = { 0 };
+
+	XMEMSET(read_key, 0xFF, sizeof(read_key));
+	ret = atcab_write_bytes_slot(0x04, 0, read_key, sizeof(read_key));
+	if (ret != ATCA_SUCCESS) {
+		printf("Failed to write key\r\n");
+		return -1;
+	}
+
+	ret = atcatlsfn_set_get_enckey(atmel_set_enc_key);
+	if (ret != ATCA_SUCCESS) {
+		printf("Failed to set enckey\r\n");
+		return -1;
+	}
+
+	return ret;
+}
+
+#endif /* WOLFSSL_ATECC508A */
+
+
+void atmel_init(void)
+{
+    if (!mAtcaInitDone) {
+#ifdef WOLFSSL_ATECC508A
+        int i;
+    #ifdef GET_REV_INFO
+        uint32_t revision = 0;
+    #endif
+        /* Init the free slot list */
+        for (i=0; i<=ATECC_MAX_SLOT; i++) {
+            if (i == 0 || i == 2 || i == 7) {
+                /* ECC Slots (mark avail) */
+                mSlotList[i] = ATECC_INVALID_SLOT;
+            }
+            else {
+                mSlotList[i] = i;
+            }
+        }
+
+        /* Initialize the CryptoAuthLib to communicate with ATECC508A */
+        atcatls_init(&cfg_ateccx08a_i2c_default);
+
+        /* Init the I2C pipe encryption key. */
+        /* Value is generated/stored during pair for the ATECC508A and stored 
+            on micro flash */
+        /* For this example its a fixed value */
+		if (atmel_init_enc_key() != ATCA_SUCCESS) {
+			printf("Failed to initialize transport key\r\n");
```

Can you assume printf() will be available?  Shouldn't we return an error here and log a message instead?